### PR TITLE
Create nlr in python

### PR DESCRIPTION
**Pull request by alexopenu**
This pr addresses issue #368. Specifically, it includes three changes:

1. Construct Network Level Reasoner in MarabouCore.createInputQuery() function.
2. Catch InputParserError that is potentially thrown during the run of MarabouCore.createInputQuery() in order to avoid a run time error with an anonymous exception.
3. Allow expressing properties on all nlr levels, not just the odd ones (e..g, only levels of type "weighted sum"). This may appear more cumbersome to some users, but also allows more flexibility - at least I find it much more convenient.

### CODE REVIEW

File: maraboupy/MarabouCore.cpp [hunk 1]
```diff
@@ -109,16 +110,29 @@ void addAbsConstraint(InputQuery& ipq, unsigned b, unsigned f){
...
+      {
+        printf( "Property: %s\n", propertyFilePathM.ascii() );
+        PropertyParser().parse( propertyFilePathM, inputQuery );
+      }
+    else
+      printf( "Property: None\n" );
+  }
+  catch(const InputParserError &e){
+        printf( "Caught an InputParserError. Code: %u. Message: %s\n", e.getCode(), e.getUserMessage() );
+        exit(1);
```
Comment (guykatzz) [inline maraboupy/MarabouCore.cpp:134]: exit(1) is a very violent way to handle an exception. How are C++ exceptions handled in other areas of the Python code?

File: src/input_parsers/PropertyParser.cpp [hunk 2]
```diff
@@ -138,7 +138,7 @@ void PropertyParser::processSingleLine( const String &line, InputQuery &inputQue
 
             auto subToken = subTokens.begin();
             ++subToken;
-            unsigned layerIndex = 2 * atoi( subToken->ascii() ) - 1;
```
Comment (guykatzz) [inline src/input_parsers/PropertyParser.cpp:141]: Yes, this makes sense. It was originally put in to maintain compatibility with the old ws_ properties, now that WS+ReLU layers are stored as two separate layers. Lets just check with @anwu1219 that this doesn't break anything of his.

File: maraboupy/MarabouCore.cpp (hunk 1 above)
Comment (alexopenu) [reply maraboupy/MarabouCore.cpp]: You are right, common practice seems to do nothing, that is, return with whatever partial results one may have; so this is what I have switched to now (there is nothing to return, but the input query may have been partly populated by this point -- which is exactly what was happening in my original example).

File: src/input_parsers/PropertyParser.cpp (hunk 2 above)
Comment (anwu1219) [reply src/input_parsers/PropertyParser.cpp:141]: Sorry somehow just saw this message. I think this should be fine on my end. Maybe it'd be good to put the convention that we are treating ws and relu as two separated layer somewhere in the README.

**Review by anwu1219 — APPROVED** [review]

File: src/input_parsers/PropertyParser.cpp (hunk 2 above)
Comment (alexopenu) [reply src/input_parsers/PropertyParser.cpp:141]: Thanks, Andrew.

**Pull request merged**